### ISSUE TITLE
Subscriptions for download and share events

### ISSUE THREAD

**Issue by AhoyLemon** [podlove/podlove-web-player#904]
As I look into the subscriptions for the web player, I see a bunch of things I can do with the switching of tabs and streaming of audio. but there's no subscriptions on anything inside of either the Download or Share buttons. 

I'm tracking play actions like this....

```javascript
podlovePlayer('#PodLovePlayer', options).then(function (store) {
      store.subscribe(() => {
        const { lastAction } = store.getState()
        if (lastAction.type == "PLAY") {
          trackEvent("listen", "play", window.location.pathname);
        }
      });
    });
```

...and I'd love to be able to extend that out with something like this...
```javascript
podlovePlayer('#PodLovePlayer', options).then(function (store) {
      store.subscribe(() => {
        const { lastAction } = store.getState()
        if (lastAction.type == "PLAY") {
          trackEvent("listen", "play", window.location.pathname);
        } else if (lastAction.type == "DOWNLOAD") {
          trackEvent("download", lastAction.payload.src, window.location.pathname);
        } else if (lastAction.type == "SHARE") {
          trackEvent("download", lastAction.payload.network, window.location.pathname);
        }
      });
    });
```

does that make sense?

**Comment by alexander-heimbuch** [comment]
Since I just expose the internal Player API, that is totally possible. You just have to listen to the `TOGGLE_TAB` action with the payload 'download' or 'share':

```javascript
if (lastAction.type === "TOGGLE_TAB" && lastAction.payload === 'download') {
   trackEvent("download", lastAction.payload.src, window.location.pathname);
}
```

Btw. here is an interactive playground for the player subscription:
https://docs.podlove.org/podlove-web-player/player-subscriptions.html

**Comment by AhoyLemon** [comment]
Yeah, I was looking at that. Currently, from the [version on the CDN](https://cdn.podlove.org/web-player/embed.js) (4.2.4), I can get payloads of the tab name when you switch tabs (`info`, `chapter`, `share` or `files`) telling me that the tab has been switched to, but no indication that the copy link or download buttons have been clicked in that tab. Is that what you meant?

**Comment by alexander-heimbuch** [comment]
Ah, yes there is right now no action attached because it triggers browser APIs (follow a link, ...). Still it is possible to proxy the events through an action :) Will consider this for the next iteration cycle since we wanted to improve the tracking abilities.

**Comment by AhoyLemon** [comment]
K! Will stick with the ChangeTab action as close enough for now.
